examples/hybrid/sphere/baroclinic_wave_utilities.jl

```diff
@@ -427,11 +427,34 @@ end
 # 0-Moment Microphysics
 
 microphysics_cache(Y, ::Nothing) = NamedTuple()
-microphysics_cache(Y, ::Microphysics0Moment) =
-    (ᶜS_ρq_tot = similar(Y.c, FT), ᶜλ = similar(Y.c, FT))
+microphysics_cache(Y, ::Microphysics0Moment) = (
+    ᶜS_ρq_tot = similar(Y.c, FT),
+    ᶜλ = similar(Y.c, FT),
+    col_integrated_precip = similar(ClimaCore.Fields.level(Y.c.ρ, 1), FT),
+)
+
+vertical∫_col(field::ClimaCore.Fields.CenterExtrudedFiniteDifferenceField) =
```

Comment:
This has come up in another case, I'll ask it here too... It looks like, because, S_rhoq_tot is on cell centers, that the vertical integral will also live at the cell center. So, the volume flux of water P (m/s) will be defined on a spherical shell of radius R_earth+dz/2. Then the net water volume lost to atmos in a step dt is P*4*pi*(R_earth+dz/2)^2*dt.

However, the land boundary flux is defined on the face of the lowest cell - a spherical shell of radius R_earth. The same flux P would lead to a net water volume gain of P*4*pi*(R_earth)^2*dt. 

Should we instead InterpolateC2F the S_rhoq_tot field, and then do the vertical integral?

Reply:
> just to check - the integral over the face field and the integral over the center field should give the same # in the limit of dz -> 0, right? 

Yes, it should. That would probably be a good convergence test in ClimaCore.

> This code is implemented because you want the result to either live on the cell center or the cell face of the lowest cell?

The two user-facing methods are needed because cell center and cell face data live on different spaces, and cell center levels are indexed with `Int`, while cell face are indexed with `PlusHalf`s. The result should live on the same space (which is arguably wrong?). Generally it depends on the integration quadrature, but I think it's not too big of an issue so long it's reduced to a single planar field.